### PR TITLE
POC multiple kernels in one xclbin

### PR DESCRIPTION
**Pull request by fifield**
This adds `--xclbin-input=input.xclbin` flag to aiecc.py. If this flag is set then `--input input.xclbin` is passed to xclbinutil during xclbin creation, and the new kernel (pdi) is appended to the existing xclbin, creating a new xclbin (specified by `--xclbin-name=output.xclbin` as before).

### CODE REVIEW

File: python/compiler/aiecc/cl_arguments.py [hunk 1]
```diff
@@ -239,6 +239,12 @@ def parse_args(args=None):
         const=True,
         help="Generate xclbin",
     )
+    parser.add_argument(
+        "--xclbin-input-name",
```
Comment (stephenneuendorffer) [inline python/compiler/aiecc/cl_arguments.py:243]: maybe just "--xclbin-input"?

File: test/npu-xrt/add_one_two/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+if (NOT WSL)
+    set(BOOST_ROOT /usr/include/boost CACHE STRING "Path to Boost install")
+    set(XRT_INC_DIR /opt/xilinx/xrt/include CACHE STRING "Path to XRT cloned repo")
+    set(XRT_LIB_DIR /opt/xilinx/xrt/lib CACHE STRING "Path to xrt_coreutil.lib")
+else()
+    set(BOOST_ROOT C:/Technical/thirdParty/boost_1_83_0 CACHE STRING "Path to Boost install")
+    set(XRT_INC_DIR C:/Technical/XRT/src/runtime_src/core/include CACHE STRING "Path to XRT cloned repo")
+    set(XRT_LIB_DIR C:/Technical/xrtNPUfromDLL CACHE STRING "Path to xrt_coreutil.lib")
+endif()
```
Comment (stephenneuendorffer) [inline test/npu-xrt/add_one_two/CMakeLists.txt:26]: These defaults seem sketchy, especially the windows ones.

File: test/npu-xrt/add_one_two/Makefile [hunk 3]
```diff
@@ -0,0 +1,40 @@
+include ../makefile-common
+
+targetname = addOneObjfifo
+
+all: build2/final.xclbin addOneObjfifo.exe
+
+build1/first.xclbin: aie1.mlir
+	mkdir -p ${@D}
+	cd ${@D} && aiecc.py \
+				--xclbin-kernel-name=ADDONE --xclbin-kernel-id=0x901 --xclbin-instance-name=ADDONEINST\
```
Comment (stephenneuendorffer) [inline test/npu-xrt/add_one_two/Makefile:10]: I'm skeptical of all the xclbin specific command line options, but I guess this is good enough for now.

**Review by stephenneuendorffer — APPROVED** [review]

**Comment by stephenneuendorffer** [comment]
Thanks for this, BTW.  I hadn't thought of an incremental approach like this.

File: test/npu-xrt/add_one_two/CMakeLists.txt (hunk 2 above)
Comment (fifield) [reply test/npu-xrt/add_one_two/CMakeLists.txt:26]: I'll need to defer to others on where this comes from. I just `cp -r add_one_objFifo add_one_two` to get started :)

File: test/npu-xrt/add_one_two/Makefile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	cp _build/${targetname}.exe $@
+else
+	cp _build/${targetname} $@ 
+endif
+
+run: ${targetname}.exe build2/final.xclbin
+	${powershell} ./$< -v 1 -x build2/final.xclbin -i build2/insts.txt -k ADDTWO
+
+run1: ${targetname}.exe build1/first.xclbin
+	${powershell} ./$< -v 1 -x build1/first.xclbin -i build1/insts.txt -k ADDONE
```
Comment (makslevental) [inline test/npu-xrt/add_one_two/Makefile:37]: this is wrong - `test.cpp` doesn't take the `-k` param and actually executes both kernels

File: python/compiler/aiecc/cl_arguments.py (hunk 1 above)
Comment (fifield) [reply python/compiler/aiecc/cl_arguments.py]: sure, done.

File: test/npu-xrt/add_one_two/CMakeLists.txt (hunk 2 above)
Comment (fifield) [reply test/npu-xrt/add_one_two/CMakeLists.txt]: removed

File: test/npu-xrt/add_one_two/Makefile (hunk 3 above)
Comment (fifield) [reply test/npu-xrt/add_one_two/Makefile]: removed

File: test/npu-xrt/add_one_two/Makefile (hunk 4 above)
Comment (fifield) [reply test/npu-xrt/add_one_two/Makefile]: removed

**Pull request merged**